### DISCUSSION TITLE
Automate - query rows not returning the MSSQL row ID

### DISCUSSION THREAD

**Discussion by benaunt** [Budibase/budibase#11693] [Help]
Hi,

I am trying to send data between Budibase and n8nio to perform some more advanced automation.

Within budibase i have a rest API query, that i am using within automate to send data from a query row action to an n8nio webhook.

I noticed that the query row action does not return the primary key field "id" column. To try get around this i made a calculated column within budibase that references the id column. Unfortunately, this also appears blank in the data section.

I ended up having to do a simple SQL query and pointing everything to that, but i think this sounds like a bug or missing functionality on both the calculated columns and the query row action in automate.

Thanks
Ben

**Comment by melohagan** [comment]
Hey Ben,

There's no reason I can think of that the query rows action step wouldn't return the primary key of a SQL table.

Could you provide some screenshots of you automation and table schema - I may be able to spot something.

**Reply by ConorWebb96** [reply]
Just going to add a bit to this, the _id is usually the primary key. However you would likely want to decode this before sending it off elsewhere. Below is a function I've use to achieve this;

```
function decodeID(encodedID) {
  return parseInt(decodeURIComponent(encodedID).replace(/\D/g, ""), 10);
}

return decodeID("%5B153%5D");
```
output - %5B153%5D -> 153

I hope this helps.
